lib/amazon-pricing/helpers/instance-type.rb

```diff
@@ -154,10 +155,23 @@ def api_name_to_nf(name)
         size_to_nf[name.split('.').last]
       end
 
+      # note: the next smaller type may _not_ be supported for a given family
+      #  so this returns the next logical/possible smaller type, but not necessarily
+      #  the next valid type
```

Comment:
This is an dangerous (and unsuitable) public API.
Either take a size (large) and return the next smaller size (medium).
Or take a valid type (c5.xlarge) and return a valid smaller type (c5.large).

Reply:
Where is this used?

Reply:
I agree, this API isn't appropriate.  We don't use this API, but removing it should mean a `major` version bump (and that requires us to change all our deployables dependencies at the same time).  I decided to defer this to a more convenient time in our schedule, and noted that this API in my comment as problematic, as a way to notify users (implicitly) of a deprecation.